Iridium driver and support

dagar opened the pull request:
@LorenzMeier let's use this PR to discuss the design and implement.

TODO
- initial support for Rockblock Mk2
- SYS_COMPANION param to start mavlink on /dev/iridium
- what about mavlink heartbeat? Disable default messages (https://github.com/PX4/Firmware/blob/master/src/modules/mavlink/mavlink_main.cpp#L1912)
- Mavlink HIGH_LATENCY message https://github.com/mavlink/mavlink/pull/634
- design considerations for extending support to other modems
- consider splitting the driver like gpsdriver for easy QGC support
- uorb message for logging Iridium status
- different fail safe behaviour for this type of link
- option to only transmit if primary link is lost

LorenzMeier commented:
Message is in, so MAVLink submodules can be updated.

LorenzMeier commented:
We probably should do a quick call to work out the best approach and how to get messages from the web interface into a GCS.

dagar commented:
I'll have the modem later this week. I'll get the driver to the point it
sends something, and then let's see how to pull.

Should we continue overloading the use of SYS_COMPANION for configuring
telem2?

On Tue, Oct 11, 2016 at 2:59 AM, Lorenz Meier <notifications@github.com>
wrote:

> We probably should do a quick call to work out the best approach and how
> to get messages from the web interface into a GCS.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/PX4/Firmware/pull/5633#issuecomment-252829503>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAFK6FLPvXjgcIZXhDBimytTS_KI6ejPks5qyzPfgaJpZM4KTGwJ>
> .
>

bartosz-wawrzacz commented:
Actually I've been working on this for a while. The pixhawk driver is close to being finished. I can share at the end of the week. To make it easy to integrate into the system it's implemented as a CDev driver, so that the mavlink app can open it directly just like a UART port.

As for the integration into QGC - I am using a python script running on a fixed-IP server, which receives the messages from Iridium (or rock7 in this case) and has a TCP interface for QGC. Saves the hassle of changing the delivery destination every time and also buffers the messages in the case of internet connection loss on the GS.

dagar commented:
Thanks Bartosz that would be great!

On Thu, Oct 13, 2016 at 4:37 AM, Bartosz Wawrzacz <notifications@github.com>
wrote:

> Actually I've been working on this for a while. The pixhawk driver is
> close to being finished. I can share at the end of the week. To make it
> easy to integrate into the system it's implemented as a CDev driver, so
> that the mavlink app can open it directly just like a UART port.
>
> As for the integration into QGC - I am using a python script running on a
> fixed-IP server, which receives the messages from Iridium (or rock7 in this
> case) and has a TCP interface for QGC. Saves the hassle of changing the
> delivery destination every time and also buffers the messages in the case
> of internet connection loss on the GS.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/PX4/Firmware/pull/5633#issuecomment-253450411>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAFK6F_COwQLeE4uMzKDKfOtIqOKc9mtks5qze3VgaJpZM4KTGwJ>
> .
>

LorenzMeier commented:
Jenkins add to whitelist

bartosz-wawrzacz commented:
rockBlock driver here:
https://github.com/bartosz-wawrzacz/Firmware/tree/satcom
Sending messages both ways works. Not tested with the mavlink app yet.

dagar commented:
@bartosz-wawrzacz I pulled your driver into the iridium branch off of the main repo. I'll continue implementing the mavlink and supporting changes here. If you have further driver changes you could open a pull request to the iridium branch.

dagar commented:
@LorenzMeier @bartosz-wawrzacz there's a lot more that could be done here, but this is working.

 changes to @bartosz-wawrzacz's driver
  - On startup rockblock disable echo and flow control
  - fill in poll, IOCTL read, write and fixed a write bug so that mavlink can read/write /dev/iridium
  - added telemetry_status logging
  - limited each transmission to 50 bytes (TODO: parameterize)
  - renamed to IridiumSBD (seemed more appropriate)

 TODO:
  - add parameter to only transmit while armed, !landed, datalink loss
  - parameters for transmit size
  - testing of simultaneous send/receive (ring alerts?)

LorenzMeier commented on .gitmodules:
```diff
@@ -1,9 +1,9 @@
 [submodule "mavlink/include/mavlink/v1.0"]
 	path = mavlink/include/mavlink/v1.0
-	url = https://github.com/mavlink/c_library_v1.git
+	url = https://github.com/dagar/c_library_v1.git
```
Need to change this back

LorenzMeier commented on .gitmodules:
```diff
@@ -1,9 +1,9 @@
 [submodule "mavlink/include/mavlink/v2.0"]
 	path = mavlink/include/mavlink/v2.0
-	url = https://github.com/mavlink/c_library_v2.git
+	url = https://github.com/dagar/c_library_v2.git
```
Here too

LorenzMeier commented on cmake/configs/nuttx_px4fmu-v2_default.cmake:
```diff
@@ -48,6 +48,7 @@ set(config_module_list
 	drivers/bst
 	drivers/snapdragon_rc_pwm
 	drivers/lis3mdl
+	drivers/iridiumsbd
```
Should we try to put this into the FMUv3 config?

LorenzMeier commented:
@dagar This could go now onto master - can you address the points from the review?

bartosz-wawrzacz commented:
I'll have some changes in the upcoming week as well.

dagar commented:
If it's not too late I was hoping to make a couple minor mavlink HIGH_LATENCY changes first. There were some good points in the mavlink PR. We could drop a couple fields, combine GPS, and squeeze a smaller timestamp field back in.

LorenzMeier commented:
This will anyway not make 1.5.1, so feel free to change.

LorenzMeier commented:
@bartosz-wawrzacz @dagar Can you rebase and include your changes some time?

dagar commented:
Fixed submodules, added to fmu-v3, and rebased.
